### PR TITLE
Runscanner deployment fixes

### PR DESCRIPTION
**Pull request by apmasell**

### CODE REVIEW

File: runscanner/src/main/java/uk/ac/bbsrc/tgac/miso/runscanner/processors/StandardIllumina.java [hunk 1]
```diff
@@ -71,17 +78,18 @@ public NotificationDto process(File runDirectory, TimeZone tz) throws IOExceptio
     // The Illumina library can't distinguish between a failed run and one that either finished or is still going. Scan the logs, if
     // available to determine if the run failed.
     File rtaLogDir = new File(runDirectory, "/Data/RTALogs");
```
Comment (sqwci) [inline runscanner/src/main/java/uk/ac/bbsrc/tgac/miso/runscanner/processors/StandardIllumina.java]: MINOR Refactor your code to get this URI from a customizable parameter. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1075)

File: runscanner/src/main/java/uk/ac/bbsrc/tgac/miso/runscanner/processors/StandardIllumina.java [hunk 2]
```diff
@@ -43,10 +45,15 @@ public StandardIllumina(Builder builder) {
   }
 
   @Override
+  public Stream<File> getRunsFromRoot(File root) {
+    return Arrays.stream(root.listFiles(File::isDirectory));
+  }
+
+  @Override
   public NotificationDto process(File runDirectory, TimeZone tz) throws IOException {
     // Call the C++ program to do the real work and write a notification DTO to standard output. The C++ object has no direct binding to the
```
Comment (sqwci) [inline runscanner/src/main/java/uk/ac/bbsrc/tgac/miso/runscanner/processors/StandardIllumina.java]: MAJOR This block of commented-out lines of code should be removed. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3ACommentedOutCodeLine)

**Comment by sqwci** [comment]
SonarQube analysis reported 6 issues
* ![MAJOR][MAJOR] 5 major
* ![MINOR][MINOR] 1 minor

Watch the comments in this conversation to review them.

#### 4 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [PacBioProcessor.java#L65](https://github.com/TGAC/miso-lims/blob/d3195a672c78a24905acf4a0bb158c6b03a86541/runscanner/src/main/java/uk/ac/bbsrc/tgac/miso/runscanner/processors/PacBioProcessor.java#L65): Rename field "status" [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1700)
1. ![MAJOR][MAJOR] [PacBioProcessor.java#L195](https://github.com/TGAC/miso-lims/blob/d3195a672c78a24905acf4a0bb158c6b03a86541/runscanner/src/main/java/uk/ac/bbsrc/tgac/miso/runscanner/processors/PacBioProcessor.java#L195): Extract the assignment out of this expression. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AAssignmentInSubExpressionCheck)
1. ![MAJOR][MAJOR] [StandardIllumina.java#L39](https://github.com/TGAC/miso-lims/blob/d3195a672c78a24905acf4a0bb158c6b03a86541/runscanner/src/main/java/uk/ac/bbsrc/tgac/miso/runscanner/processors/StandardIllumina.java#L39): Remove this unused method parameter "parameters". [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1172)
1. ![MAJOR][MAJOR] [StandardIllumina.java#L64](https://github.com/TGAC/miso-lims/blob/d3195a672c78a24905acf4a0bb158c6b03a86541/runscanner/src/main/java/uk/ac/bbsrc/tgac/miso/runscanner/processors/StandardIllumina.java#L64): Either re-interrupt this method or rethrow the "InterruptedException". [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2142)

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by apmasell** [comment]
Once again, I think everything Sonar has to say is unhelpful.

**Comment by sqwci** [comment]
SonarQube analysis reported 6 issues
* ![MAJOR][MAJOR] 5 major
* ![MINOR][MINOR] 1 minor

Watch the comments in this conversation to review them.

#### 4 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [PacBioProcessor.java#L65](https://github.com/TGAC/miso-lims/blob/cec6dd6586a9ac2d8691c938a32dd958707e8fd5/runscanner/src/main/java/uk/ac/bbsrc/tgac/miso/runscanner/processors/PacBioProcessor.java#L65): Rename field "status" [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1700)
1. ![MAJOR][MAJOR] [PacBioProcessor.java#L195](https://github.com/TGAC/miso-lims/blob/cec6dd6586a9ac2d8691c938a32dd958707e8fd5/runscanner/src/main/java/uk/ac/bbsrc/tgac/miso/runscanner/processors/PacBioProcessor.java#L195): Extract the assignment out of this expression. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AAssignmentInSubExpressionCheck)
1. ![MAJOR][MAJOR] [StandardIllumina.java#L39](https://github.com/TGAC/miso-lims/blob/cec6dd6586a9ac2d8691c938a32dd958707e8fd5/runscanner/src/main/java/uk/ac/bbsrc/tgac/miso/runscanner/processors/StandardIllumina.java#L39): Remove this unused method parameter "parameters". [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1172)
1. ![MAJOR][MAJOR] [StandardIllumina.java#L64](https://github.com/TGAC/miso-lims/blob/cec6dd6586a9ac2d8691c938a32dd958707e8fd5/runscanner/src/main/java/uk/ac/bbsrc/tgac/miso/runscanner/processors/StandardIllumina.java#L64): Either re-interrupt this method or rethrow the "InterruptedException". [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2142)

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by sqwci** [comment]
SonarQube analysis reported 8 issues
* ![CRITICAL][CRITICAL] 1 critical
* ![MAJOR][MAJOR] 6 major
* ![MINOR][MINOR] 1 minor

Watch the comments in this conversation to review them.

#### 6 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![CRITICAL][CRITICAL] [NotificationDto.java#L131](https://github.com/TGAC/miso-lims/blob/2aa2d7e65f9c49eceba084e1aa63d0602c90725e/miso-dto/src/main/java/uk/ac/bbsrc/tgac/miso/dto/NotificationDto.java#L131): Rename this constant name to match the regular expression '^[A-Z][A-Z0-9]*(_[A-Z0-9]+)*$'. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS00115)
1. ![MAJOR][MAJOR] [NotificationDto.java#L183](https://github.com/TGAC/miso-lims/blob/2aa2d7e65f9c49eceba084e1aa63d0602c90725e/miso-dto/src/main/java/uk/ac/bbsrc/tgac/miso/dto/NotificationDto.java#L183): Remove this unused method parameter "lane". [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1172)
1. ![MAJOR][MAJOR] [PacBioProcessor.java#L65](https://github.com/TGAC/miso-lims/blob/2aa2d7e65f9c49eceba084e1aa63d0602c90725e/runscanner/src/main/java/uk/ac/bbsrc/tgac/miso/runscanner/processors/PacBioProcessor.java#L65): Rename field "status" [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1700)
1. ![MAJOR][MAJOR] [PacBioProcessor.java#L195](https://github.com/TGAC/miso-lims/blob/2aa2d7e65f9c49eceba084e1aa63d0602c90725e/runscanner/src/main/java/uk/ac/bbsrc/tgac/miso/runscanner/processors/PacBioProcessor.java#L195): Extract the assignment out of this expression. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AAssignmentInSubExpressionCheck)
1. ![MAJOR][MAJOR] [StandardIllumina.java#L39](https://github.com/TGAC/miso-lims/blob/2aa2d7e65f9c49eceba084e1aa63d0602c90725e/runscanner/src/main/java/uk/ac/bbsrc/tgac/miso/runscanner/processors/StandardIllumina.java#L39): Remove this unused method parameter "parameters". [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1172)
1. ![MAJOR][MAJOR] [StandardIllumina.java#L64](https://github.com/TGAC/miso-lims/blob/2aa2d7e65f9c49eceba084e1aa63d0602c90725e/runscanner/src/main/java/uk/ac/bbsrc/tgac/miso/runscanner/processors/StandardIllumina.java#L64): Either re-interrupt this method or rethrow the "InterruptedException". [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2142)

[CRITICAL]: https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by sqwci** [comment]
SonarQube analysis reported 8 issues
* ![CRITICAL][CRITICAL] 1 critical
* ![MAJOR][MAJOR] 6 major
* ![MINOR][MINOR] 1 minor

Watch the comments in this conversation to review them.

#### 6 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![CRITICAL][CRITICAL] [NotificationDto.java#L131](https://github.com/TGAC/miso-lims/blob/b0c496915e6a3127fad90ce99638db3caed03478/miso-dto/src/main/java/uk/ac/bbsrc/tgac/miso/dto/NotificationDto.java#L131): Rename this constant name to match the regular expression '^[A-Z][A-Z0-9]*(_[A-Z0-9]+)*$'. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS00115)
1. ![MAJOR][MAJOR] [NotificationDto.java#L183](https://github.com/TGAC/miso-lims/blob/b0c496915e6a3127fad90ce99638db3caed03478/miso-dto/src/main/java/uk/ac/bbsrc/tgac/miso/dto/NotificationDto.java#L183): Remove this unused method parameter "lane". [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1172)
1. ![MAJOR][MAJOR] [PacBioProcessor.java#L65](https://github.com/TGAC/miso-lims/blob/b0c496915e6a3127fad90ce99638db3caed03478/runscanner/src/main/java/uk/ac/bbsrc/tgac/miso/runscanner/processors/PacBioProcessor.java#L65): Rename field "status" [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1700)
1. ![MAJOR][MAJOR] [PacBioProcessor.java#L195](https://github.com/TGAC/miso-lims/blob/b0c496915e6a3127fad90ce99638db3caed03478/runscanner/src/main/java/uk/ac/bbsrc/tgac/miso/runscanner/processors/PacBioProcessor.java#L195): Extract the assignment out of this expression. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AAssignmentInSubExpressionCheck)
1. ![MAJOR][MAJOR] [StandardIllumina.java#L39](https://github.com/TGAC/miso-lims/blob/b0c496915e6a3127fad90ce99638db3caed03478/runscanner/src/main/java/uk/ac/bbsrc/tgac/miso/runscanner/processors/StandardIllumina.java#L39): Remove this unused method parameter "parameters". [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1172)
1. ![MAJOR][MAJOR] [StandardIllumina.java#L64](https://github.com/TGAC/miso-lims/blob/b0c496915e6a3127fad90ce99638db3caed03478/runscanner/src/main/java/uk/ac/bbsrc/tgac/miso/runscanner/processors/StandardIllumina.java#L64): Either re-interrupt this method or rethrow the "InterruptedException". [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2142)

[CRITICAL]: https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

File: miso-dto/src/main/java/uk/ac/bbsrc/tgac/miso/dto/ProgressiveResponseDto.java [hunk 3]
```diff
@@ -32,6 +33,14 @@ public long getToken() {
     return updates;
   }
 
+  public boolean isAgain() {
```
Comment (djcooke) [inline miso-dto/src/main/java/uk/ac/bbsrc/tgac/miso/dto/ProgressiveResponseDto.java]: Seems to indicate that more new data is available than included, but 'isAgain' sounds like it's a repeat of old data. Please rename to clarify meaning

File: miso-web/src/main/java/uk/ac/bbsrc/tgac/miso/webapp/service/RunScannerClient.java [hunk 4]
```diff
@@ -114,7 +114,7 @@ public void scheduler() {
     }
   }
 
-  @Value("${miso.runscanner.urls}")
+  @Value("${miso.runscanner.urls:}")
```
Comment (djcooke) [inline miso-web/src/main/java/uk/ac/bbsrc/tgac/miso/webapp/service/RunScannerClient.java]: :100:

File: miso-dto/src/main/java/uk/ac/bbsrc/tgac/miso/dto/ProgressiveResponseDto.java (hunk 3 above)
Comment (apmasell) [inline miso-dto/src/main/java/uk/ac/bbsrc/tgac/miso/dto/ProgressiveResponseDto.java]: I guess this is terminology from UNIX where `EAGAIN` means “repeat your request for more”. What do you want it to be called?

Comment (djcooke) [inline miso-dto/src/main/java/uk/ac/bbsrc/tgac/miso/dto/ProgressiveResponseDto.java]: `isMoreAvailable`, `hasMoreData`, `toBeContinued`, `butThatsNotAll`, etc

**Pull request merged**